src/lib/core/theming/theme-check.ts

```diff
@@ -0,0 +1,40 @@
+import {NgModule, isDevMode} from '@angular/core';
+
+/** Whether the theme presence has already been checked. */
+let hasBeenChecked = false;
+
+/**
+ * Module that verifies that the user has loaded the core theming file,
+ * without which most Material module won't work as expected.
+ *
+ * Note on testing methodology: A more efficient way to check for the theme
+ * would be to loop through the `document.styleSheets`, however most browsers
+ * don't expose the stylesheet rules, if the file was loaded from another domain.
+ * This would trigger false positives if the theme is being loaded from a CDN.
+ *
+ * @docs-private
+ */
+@NgModule()
+export class MdThemeCheckModule {
+  constructor() {
+    if (hasBeenChecked || typeof document === 'undefined' || !isDevMode()) {
+      return;
+    }
+
+    let testElement = document.createElement('div');
+
+    testElement.classList.add('md-theme-loaded-marker');
+    document.body.appendChild(testElement);
+
+    if (getComputedStyle(testElement).display !== 'none') {
```

Comment:
What if instead we did something like
```scss
.mat-theme-load-test {
  width: 500px;
}
```

Then we create two elements (position absolute, min heights/width 1px), append them to body, and add the class to one of them then see if they have the same/different offsetWidth?